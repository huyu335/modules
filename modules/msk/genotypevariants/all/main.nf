process GENOTYPEVARIANTS_ALL {
    tag "$meta.id"
    label 'process_single'


    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/genotype_variants:sha-60b7f8bc':
        'ghcr.io/msk-access/genotype_variants:sha-60b7f8bc' }"

    input:
    tuple val(meta), path(bam)
    tuple val(meta), path(maf)
    path  fasta
    path  fai

    output:
    tuple val(meta), path("*.maf"), emit: maf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def sample = "${meta.sample}"
    def patient = "${meta.patient}"
    // TODO nf-core: If the tool supports multi-threading then you MUST provide the appropriate parameter
    //               using the Nextflow "task" variable e.g. "--threads $task.cpus"
    """
    genotype_variants small_variants all \\
        -i ${maf} \\ 
        -r ${fasta} \\
        -g /usr/local/bin/GetBaseCountsMultiSample \\
        -p ${patient} \\
        -b ${bam} \\
        -s ${sample} \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genotypevariants: \$(genotype_variants --version)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    // TODO nf-core: A stub section should mimic the execution of the original module as best as possible
    //               Have a look at the following examples:
    //               Simple example: https://github.com/nf-core/modules/blob/818474a292b4860ae8ff88e149fbcda68814114d/modules/nf-core/bcftools/annotate/main.nf#L47-L63
    //               Complex example: https://github.com/nf-core/modules/blob/818474a292b4860ae8ff88e149fbcda68814114d/modules/nf-core/bedtools/split/main.nf#L38-L54
    """
    touch ${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        genotypevariants: \$(samtools --version |& sed '1!d ; s/samtools //')
    END_VERSIONS
    """
}
