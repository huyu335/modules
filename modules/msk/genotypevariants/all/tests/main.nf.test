// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
// nf-core modules test genotypevariants/all
nextflow_process {

    name "Test Process GENOTYPEVARIANTS_ALL"
    script "../main.nf"
    process "GENOTYPEVARIANTS_ALL"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "genotypevariants"
    tag "genotypevariants/all"


    test("sarscov2 - bam, msk - maf") {
        when {
            process {
                """
                input[0] = [
                            [ id:'test', sample:'197', patient:'c-id'], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam_bai'], checkIfExists: true)
                        ]
                input[1] = file(params.test_data_msk['bams']['chr22maf'], checkIfExists: true)
                input[2] = file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true)
                input[3] = file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sarscov2 - bam - stub") {

        config './nextflow.config'
        options "-stub"

        when {
            process {
                """
                input[0] = [
                            [ id:'test', sample:'197', patient:'c-id'], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_single_end_sorted_bam_bai'], checkIfExists: true)
                        ]
                input[1] = []
                input[2] = []
                input[3] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
