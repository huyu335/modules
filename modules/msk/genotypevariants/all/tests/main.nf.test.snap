{
    "sarscov2 - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        "197.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,bde4f933543cca4f05d70a7f0347e9bf"
                ],
                "maf": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        "197.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bde4f933543cca4f05d70a7f0347e9bf"
                ]
            }
        ],
        "timestamp": "2024-01-31T10:44:42.455827"
    },
    "sarscov2 - bam, msk - maf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        [
                            "197-ORG-STD_genotyped.maf:md5,8a140bbb7af1c291e8597866fb9d6bb9",
                            "197-STANDARD_genotyped.maf:md5,6c0649cfd70ceb0587827f48bf93caf7"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,bde4f933543cca4f05d70a7f0347e9bf"
                ],
                "maf": [
                    [
                        {
                            "id": "197",
                            "patient": "c-id"
                        },
                        [
                            "197-ORG-STD_genotyped.maf:md5,8a140bbb7af1c291e8597866fb9d6bb9",
                            "197-STANDARD_genotyped.maf:md5,6c0649cfd70ceb0587827f48bf93caf7"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,bde4f933543cca4f05d70a7f0347e9bf"
                ]
            }
        ],
        "timestamp": "2024-01-31T10:44:36.646565"
    }
}